### PR TITLE
Added Postman Sort Algorithm in C and Python 3.

### LINKED ISSUE #1412: Postman Sort in C and Python
Please assign this to me

### PR DESCRIPTION
**Pull request by jhasaurabh312**
Fixes #1412 

#### Checklist:
- [x] 4 space indentation.
- [x]  Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Postman Sort in C
- Postman Sort in Python

#### Languages Used:
- C
- Python 3

#### Files Added:
- Postman_Sort/Postman_sort.c
- Postman_Sort/Postman_sort.py

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by jhasaurabh312** [comment]
I have added the Postman Sort Algorithm in C and Python 3. Please review and merge the changes if they are correct. Thanks !!!

File: Postman_Sort/Postman_sort.c [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        for (j = i + 1; j < n; j++)
+        {
+            if (array1[i] > array1[j])
+            {
+
+                temp = array1[i];          // swapping of elements
+                array1[i] = array1[j];
+                array1[j] = temp;
+
+
```
Comment (jainaman224) [inline Postman_Sort/Postman_sort.c]: Don't add unnecessary multi new lines.

File: Postman_Sort/Postman_sort.c [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+}
+
+
+
+int main()
+{
+
+    int t1, t, n = 1;
```
Comment (jainaman224) [inline Postman_Sort/Postman_sort.c]: Use better variable name then t1.

File: Postman_Sort/Postman_sort.c [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    printf("Enter the size of array : ");
+    scanf("%d", &count);  // size of array
+
+    printf("Enter the elements of array : ");
+    for (i = 0; i < count; i++)
+
+    {
+
```
Comment (jainaman224) [inline Postman_Sort/Postman_sort.c]: Remove so many new lines.

Comment (jhasaurabh312) [reply Postman_Sort/Postman_sort.c]: Okk I will do the suggested changes and make a pr! !!

**Comment by jhasaurabh312** [comment]
Hey I have done all the changes i had to make. Please review the PR. Thanks !!!

File: Postman_Sort/Postman_sort.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
+#include <stdio.h>
+int array[100], array1[100];
```
Comment (jainaman224) [inline Postman_Sort/Postman_sort.c]: Don't use global variables like this.

File: Postman_Sort/Postman_sort.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    list[pos1], list[pos2] = list[pos2], list[pos1]
+    return list
+
+def swapPositionsMOD(list, pos1, pos2): 
+
+    list[pos1], list[pos2] = list[pos2]%10, list[pos1]%10
+    return list
+
+
+count = int(input("Size of the array : "))
```
Comment (jainaman224) [inline Postman_Sort/Postman_sort.py]: Wrap this in function ideally named main.

Comment (jainaman224) [reply Postman_Sort/Postman_sort.py]: It was hard for me to see from where code started to execute.

File: Postman_Sort/Postman_sort.c [hunk 6]
```diff
@@ -0,0 +1,109 @@
+#include <stdio.h>
+int i, j, temp, max, count, maxdigits = 0, c = 0;
```
Comment (jainaman224) [inline Postman_Sort/Postman_sort.c]: Don't use global variables.

File: Postman_Sort/Postman_sort.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
+'''Size of the array : 5
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: sample should be at end.

File: Postman_Sort/Postman_sort.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    print("Enter the elements : ")
+    for i in range(0, count):
+        ele = int(input())
+        array.append(ele) # adding the element
+        array1.append(ele) # adding the element
+
+    print("Input Array without Sorting : ")
+    print(array)
+
+    maxdigits=0
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+
+def swapPositions(list, pos1, pos2):
+
+    list[pos1], list[pos2] = list[pos2], list[pos1]
+    return list
+
+def swapPositionsMOD(list, pos1, pos2):
+
+    list[pos1], list[pos2] = list[pos2]%10, list[pos1]%10
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+            k=i
+            z=j
+
+            for a in range(k,z-1) :
+                for b in range(a+1,z) :
+                    if array1[a] > array1[b] :
+                        swapPositions(array1,a,b)
+                        swapPositionsMOD(array,a,b)
+            i = j
+        n = n/10
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        while i<count :
+            t1 = int(array[i] / n)
+            j=i+1
+            while j<count and t1==int(array[j]/n) :
+                j = j+1
+
+            k=i
+            z=j
+
+            for a in range(k,z-1) :
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: add space after ,

File: Postman_Sort/Postman_sort.py [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+                for b in range(a+1,z) :
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: add space after ,

File: Postman_Sort/Postman_sort.py [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+                    if array1[a] > array1[b] :
+                        swapPositions(array1,a,b)
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: add space after ,

File: Postman_Sort/Postman_sort.py [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+                        swapPositionsMOD(array,a,b)
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: add space after ,

File: Postman_Sort/Postman_sort.py [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+        swapPositions(array,t,i)
+        swapPositions(array1,t,i)
+
+    while n >= 1 :
+        i=0
+        while i<count :
+            t1 = int(array[i] / n)
+            j=i+1
+            while j<count and t1==int(array[j]/n) :
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: add space around the operator

File: Postman_Sort/Postman_sort.py [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+                 t = j
+
+        swapPositions(array,t,i)
+        swapPositions(array1,t,i)
+
+    while n >= 1 :
+        i=0
+        while i<count :
+            t1 = int(array[i] / n)
+            j=i+1
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+            if maxi > int(array[j]/n) :
+                 maxi = int(array[j]/n)   # greatest MSB
+                 t = j
+
+        swapPositions(array,t,i)
+        swapPositions(array1,t,i)
+
+    while n >= 1 :
+        i=0
+        while i<count :
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+        for j in range(i+1,count):
+            if maxi > int(array[j]/n) :
+                 maxi = int(array[j]/n)   # greatest MSB
+                 t = j
+
+        swapPositions(array,t,i)
+        swapPositions(array1,t,i)
+
+    while n >= 1 :
+        i=0
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+    temp=0
+    for i in range(0,count):
+        maxi = int(array[i]/n)         # MSB - Dividing by particular base
+        t = i
+        for j in range(i+1,count):
+            if maxi > int(array[j]/n) :
+                 maxi = int(array[j]/n)   # greatest MSB
+                 t = j
+
+        swapPositions(array,t,i)
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: add space after ,

File: Postman_Sort/Postman_sort.py [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+       n = n * 10
+       maxdigits -=1
+
+    temp=0
+    for i in range(0,count):
+        maxi = int(array[i]/n)         # MSB - Dividing by particular base
+        t = i
+        for j in range(i+1,count):
+            if maxi > int(array[j]/n) :
+                 maxi = int(array[j]/n)   # greatest MSB
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+
+    while maxdigits>1 :
+       n = n * 10
+       maxdigits -=1
+
+    temp=0
+    for i in range(0,count):
+        maxi = int(array[i]/n)         # MSB - Dividing by particular base
+        t = i
+        for j in range(i+1,count):
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+        c=0
+        while(t>0):
+            c += 1
+            t = int(t/10)  # Find MSB
+
+        if maxdigits < c :
+            maxdigits = c   # number of digits of a each number
+
+
+    while maxdigits>1 :
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+
+    maxdigits=0
+    n=1
+
+    for i in range(len(array)):
+        t = int(array[i])
+        c=0
+        while(t>0):
+            c += 1
+            t = int(t/10)  # Find MSB
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+    print("Input Array without Sorting : ")
+    print(array)
+
+    maxdigits=0
+    n=1
+
+    for i in range(len(array)):
+        t = int(array[i])
+        c=0
+        while(t>0):
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+
+    print("Input Array without Sorting : ")
+    print(array)
+
+    maxdigits=0
+    n=1
+
+    for i in range(len(array)):
+        t = int(array[i])
+        c=0
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

File: Postman_Sort/Postman_sort.py [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+    for i in range(0, count):
+        ele = int(input())
+        array.append(ele) # adding the element
+        array1.append(ele) # adding the element
+
+    print("Input Array without Sorting : ")
+    print(array)
+
+    maxdigits=0
+    n=1
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.py]: space around the operator

**Comment by jhasaurabh312** [comment]
I have done all the suggested changes. Please merge this PR. Thanks :)

File: Postman_Sort/Postman_sort.c [hunk 27]
```diff
@@ -0,0 +1,110 @@
...
+void arrange(int array[], int array1[], int k , int n)
+{
+    int temp = 0;
+    for (int i = k; i < n - 1; i++)
+    {
+        for (int j = i + 1; j < n; j++)
+        {
+            if (array1[i] > array1[j])
+            {
+                temp = array1[i];          // swapping of elements
```
Comment (somya-kapoor) [inline Postman_Sort/Postman_sort.c]: move comments above the statements

**Comment by somya-kapoor** [comment]
@jhasaurabh312 Are you working on it?

**Comment by somya-kapoor** [comment]
@jhasaurabh312 Due to no activity for more than 7 days, I am closing this issue.

**Pull request closed without merging**